### ISSUE TITLE
Release v0.2.0

### ISSUE THREAD

**Issue by silkentrance** [raszi/node-tmp#233]
In order to overcome the most pressing issues: #216, #213, #207, #218, #236, #237, #238, #176, #156.

I make the following proposal in order to further stabilize tmp and to get a new release out of the door as soon as possible.

- SIGINT handler (BREAKING CHANGE)
  the SIGINT handler will be removed
  The user is responsible for installing a SIGINT handler that will exit the process so that the exit handler gets called.

- EXIT handler
  The exit handler can be simplified to just a call to the internal garbage collector function.

- handler installation
  The safe handler installation had been a work around for older jest versions. With jest now using child processes and experimental worker threads, this should no longer be an issue and we can remove that code altogether. See also #229 

- CHANGELOG
  introduce a changelog which we will maintain manually for now

@raszi please have a look and tell me if that would suit you. Exposing the garbage collector as a callable has its own issues, see the comments in #216.

**Comment by silkentrance** [comment]
See PR #234

**Comment by alubbe** [comment]
@raszi any news on this? We'd love to upgrade to 0.2.0 for exceljs

**Comment by nomcopter** [comment]
When will this release? I don't even use `node-tmp` but this is breaking `SIGINT` handlers in an application of mine because a random dependency uses this which forces a complete process exit on `SIGINT`.

**Comment by silkentrance** [comment]
@raszi What about a new release on npmjs.com?

**Comment by ezze** [comment]
Still waiting for it for months. Some time ago `SIGINT` issue [have been fixed](https://github.com/raszi/node-tmp/pull/193) in `master` and then broken again. For those who is experiencing the same issue here is a temporary workaround to stick to exact working old commit:

```bash
$ rm -Rf node_modules
$ yarn remove tmp
$ yarn cache clean
$ yarn add raszi/node-tmp#ae61bb63dbc05ff66678b992a14a19f44ab899ed
```

**Comment by silkentrance** [comment]
@ezze @nomcopter @raszi just released version 0.2.0 to npmjs. Have fun.

**Comment by alubbe** [comment]
Thanks a bunch!

**Comment by ezze** [comment]
@silkentrance Thanks for doing it. Unfortunately, new release solving `SIGINT` issue introduces other breaking changes such as this one: https://github.com/raszi/node-tmp/issues/250

At the moment, we can't live with it, so I created forked npm package `tmp-sigint` that implements my workaround mentioned above for 0.1.0 (I decided to create it due to yarn 1.22.4 doesn't work well with commits breaking `yarn.lock` on each dependencies installation).

**Comment by missinglink** [comment]
Hi @silkentrance, the semantic versioning spec adopted by most `npm` modules requires that breaking changes increment the `major` version number.

In this release you've published `0.2.0` when it should really be `1.0.0`, this helps downstream consumers better control which updates they accept automatically and which require manual review.

https://semver.org/
